django/apps/registry.py

```diff
@@ -344,6 +346,76 @@ def clear_cache(self):
             for model in self.get_models(include_auto_created=True):
                 model._meta._expire_cache()
 
+    @staticmethod
+    def _model_key(model_or_name):
+        """
+        Returns an (app_label, model_name) tuple from a model or string.
+        """
+        if isinstance(model_or_name, six.string_types):
+            app_label, model_name = model_or_name.split(".")
+        else:
+            # It's actually a model class.
+            app_label = model_or_name._meta.app_label
+            model_name = model_or_name._meta.object_name
+        return app_label, model_name
+
+    def lazy_model_operation(self, function, *models_or_names, **kwargs):
```

Comment:
``**kwargs`` here don't add anything functional here - any data in keyword args could equivalently be closed over in the callback function.

However it makes things a bit less awkward: removing the need for ``partial``/``curry`` or single-use callbacks, and also allows for the introspection of pending operations in django/db/migrations/state.py.